src/ess/spectroscopy/indirect/io.py

```diff
@@ -0,0 +1,191 @@
+        NXfermi_chopper,
+        NXinstrument,
+        NXsample,
+    )
+
+    observations = events.copy()
+    ef = events.coords['final_energy']
+    observations *= sqrt(events.bins.coords['incident_energy'] / ef)
+
+    # Adding null events requires replicating all coordinates of real events.
+    # We don't necessarily use all present event coordinates, so remove any we won't use
+    targets = [
+        'energy_transfer',
+        'incident_energy',
+        'incident_wavelength',
+        'final_energy',
+    ]
+    for coord in [x for x in events.bins.coords if x not in targets]:
+        del observations.bins.coords[coord]
+    # And provide a means to calculate the rest from bin (not event) information
+    # we are sure incident_wavelength is a bin coordinate already, so can skip it.
+    graph = {
+        'incident_energy': _lambda_to_ei,
+        'energy_transfer': _ei_ef_to_en,
+    }
+    # # Adding zero-weight observations, i.e., null events, works now, but involves
+    # # a memory-copy of the data array and its bin structure. Since it isn't strictly
+    # # necessary for exporting NXspe, it can be skipped for now.
+    # observations = add_null_observations(observations, targets, graph)
+
+    # combine the per bin intensities and normalize by monitor counts
+    # Note, applying this normalization to the _events_ would require splitting
+    # the bin monitor counts between the bin events.
+    normalize_by = events.coords['monitor']  # .broadcast(sizes=observations.sizes)
+    if normalize_by.variances is not None:
+        # we need to ignore the monitor uncertainty for the time being
+        normalize_by = normalize_by.copy()
+        normalize_by.variances = None
+
+    observations = observations.hist().transform_coords(targets, graph=graph)
+
+    if observations.variances is None:
+        observations.variances = observations.values  # correct for counting statistics
+        observations.variances[observations.values == 0] = 1
+    if observations.data.variances is not None:
+        observations.data.variances[observations.data.variances == 0] = 1
+    # the transform_coords above renamed the 'incident_wavelength' dimension
+    # to 'energy_transfer' ... so we need to do the same to normalize_by or else
+    # scipp tries to broadcast when it doesn't need to.
+    observations.data = observations.data / normalize_by.rename_dims(
+        incident_wavelength='energy_transfer'
+    )
+
+    psi = observations.coords['a3']
+    polar = observations.coords['theta']
+    azimuthal = 0 * polar  # all detectors are in the horizontal plane
+    azimuthal_width = azimuthal + scalar(2.0, unit='degree')
+    polar_width = azimuthal + scalar(0.1, unit='degree')
+    distance = full(sizes=polar.sizes, unit='m', value=3.0, dtype=polar.dtype)
+    data = observations.data
+    error = 0 * observations.data.values
+    if observations.data.variances is not None:
+        error = numpy.sqrt(observations.data.variances)
```

Comment:
```suggestion
    if observations.data.variances is not None:
        error = numpy.sqrt(observations.data.variances)
    else:
        error = 0 * observations.data.values
```
To avoid the memory overhead of creating two `error` arrays.